internal/dirs/copy_test.go

```diff
@@ -0,0 +1,145 @@
+package dirs
+
+import (
+	"os"
+	"testing"
+
+	"path/filepath"
+)
+
+func TestDefault(t *testing.T) {
+	overlay := t.TempDir()
+	dest := t.TempDir()
+
+	if err := Mkdirs(filepath.Join(overlay, "content"), filepath.Join(dest, "content")); err != nil {
+		t.Fatal(err)
+	}
+
+	srcPath := filepath.Join(overlay, "content", "test_content.txt")
+	file, err :=  os.OpenFile(srcPath, os.O_CREATE|os.O_WRONLY, 0o755)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := file.WriteString("MEOWMEOW TESTING YAHH!!"); err != nil {
+		file.Close()
+		t.Fatal(err)
+	}
+
+	if err := file.Close(); err != nil {
+		t.Fatal(err)
+	}
+
+	err = OverlayDir(overlay, dest)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	destPath := filepath.Join(dest, "content", "test_content.txt")
+
+	info, err := os.Stat(srcPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if info.Mode()&os.ModeSymlink != 0 {
+		destSymPath, err := filepath.EvalSymlinks(destPath)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		srcSymPath, err := filepath.EvalSymlinks(srcPath)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if destSymPath != srcSymPath {
+			t.FailNow()
+		}
+	}
+}
+
+func TestAlreadyExists(t *testing.T) {
+	overlay := t.TempDir()
+	dest := t.TempDir()
+
+	if err := Mkdirs(filepath.Join(overlay, "content"), filepath.Join(dest, "content")); err != nil {
+		t.Fatal(err)
+	}
+
+	srcPath := filepath.Join(overlay, "content", "test_content.txt")
+	file, err :=  os.OpenFile(srcPath, os.O_CREATE|os.O_WRONLY, 0o755)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := file.WriteString("MEOWMEOW TESTING YAHH!!"); err != nil {
+		file.Close()
+		t.Fatal(err)
+	}
+
+	if err := file.Close(); err != nil {
+		t.Fatal(err)
+	}
+
+	t.Log("(1) Overlay dir: ", overlay, dest)
```

Comment:
It seems there are multiple tests in one function, please seperate them into individual tests and don't use logging. For this, you can use a generalized function to generate a directory.

Reply:
The test is to ensure the behavior is kept, but i am still concerned over having to test this ourselves for behavior that is expected in the upstream lib.

Reply:
So what should i do now? Remove log?

Reply:
Sure.

Reply:
Done